test/integration/sample_test.go

```diff
@@ -0,0 +1,79 @@
+package samples_test
+
+import (
+	"io/ioutil"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+)
+
+const (
+	setupPath    = "../setup"
+	sampleDir    = "../../"
+	sampleRegion = "us-central1"
+	sampleZone   = "us-central1-a"
+)
+
+// Retry if these errors are encountered.
+var retryErrors = map[string]string{
+	// IAM for Eventarc service agent is eventually consistent
+	".*Permission denied while using the Eventarc Service Agent.*": "Eventarc Service Agent IAM is eventually consistent",
+}
+
+func TestSamples(t *testing.T) {
+	// This initial blueprint test is to extract output info
+	// so we only have to set the env vars once.
+	setup := tft.NewTFBlueprintTest(t,
+		tft.WithTFDir(sampleDir),
+		tft.WithSetupPath(setupPath),
+	)
+	testProjectID := setup.GetTFSetupStringOutput("project_id")
+	testEnv := map[string]string{
+		"GOOGLE_PROJECT": testProjectID,
+		"GOOGLE_REGION":  sampleRegion,
+		"GOOGLE_ZONE":    sampleZone,
```

Comment:
These aren't being reused. Might as well set them here for better readability.

Reply:
Good point, removed.